### PR TITLE
refactor: Remove top-level bot loop (take two)

### PR DESCRIPTION
**Pull request by pxrl**
This change was previously merged and backed out shortly after because it was seen to result in some bot instances hanging. This is almost certainly because the original commit removed two instances of process.exit(), and replaced them with prococess.exitCode = <X>. There appear to be some instances where the bot can have some dangling promises that take a long time to resolve, or that don't resolve for some reason. Replaying the change, but without removing the process.exit() calls should be OK.

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -80,17 +59,19 @@ if (require.main === module) {
 
   run(args)
     .then(() => {
-      // eslint-disable-next-line no-process-exit
-      process.exit(0);
+      exit(0);
```
Comment (james-a-morris) [inline index.ts:62]: Thoughts on having an `exitCode` variable with a single `exit()` in the `.finally()` block?

```
let exitCode = 0;
run(args).
  .catch(() => {
      exitCode = 1;
      ...{logging}
  })
  .finally(() => {
     exit(exitCode);
  });
```

**Review by james-a-morris — COMMENTED** [review]
One nit on the structure

File: index.ts (hunk 1 above)
Comment (pxrl) [reply index.ts]: Good suggestion - implemented: https://github.com/across-protocol/relayer-v2/pull/1404/commits/d2f238f754cb26638c1a0beccf634df4a156f2f3

**Review by james-a-morris — APPROVED** [review]
LGTM pending the linter fix

**Pull request merged**